src/ft2font.h

```diff
@@ -144,6 +145,7 @@ class FT2Font
     FT_Vector pen;    /* untransformed origin  */
     std::vector<FT_Glyph> glyphs;
     std::vector<FT2Font *> fallbacks;
+    std::set<FT_String*> family_names;
```

Comment:
Rather than adding state, can we add a parameter to `load_char_with_fallback` to accumulate the tried (but failed) names.  It can be allocated by code that first calls the recursive function and `load_char` then passed through to all of the generations.  Then again in `load_char` you can either warn or not and then clear the list.

Reply:
Please try that again, I am not clear how local state is going to be messier than global state.

As the fallback fonts are also `F2Font` objects they can technically (but I don't think we provide a way from Python to do so) contain their own fallback lists.  Passing an accumulator / cache through the call should work fine in that case if we ever do enable that.